students/larry_mburu/Lesson07/html_render.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+
+# This is the framework for the base class
+class Element(object):
+    
+    indent = '   '
+    tag = 'html'
+
+    def __init__(self, content=None, **attrs):
+        if content is not None:
+            self.content = [content]
+        else: 
+            self.content = []
+
+        self.attributes = attrs # element attributes
+
+    def append(self, new_content):
+        self.content.append(new_content)
+
+    def render(self, out_file, cur_ind=""):
+        if cur_ind: 
+            out_file.write(f"{self.indent}{self.render_opening_tag()}")
+        else:
+            out_file.write(f"{self.render_opening_tag()}")
+        out_file.write("\n")
+        for content in self.content:
+            try: 
+                out_file.write(f"{self.indent}" + content)
+            except TypeError: 
+                content.render(out_file)
+            out_file.write("\n")
+        if cur_ind:
+            out_file.write(f"{self.indent}</{self.tag}>")
+        else:
+            out_file.write(f"</{self.tag}>")
+
+    def render_opening_tag(self):
+        """
+        returns formatted opening tags if given attributes
+        """
+        self.compose = f"<{self.tag}>" # self.compose - to build out element
+        if len(self.attributes) == 0:  # zero attrs given. don't change tag structure!
+            return self.compose
+        else:
+            self.compose = f"<{self.tag}" # one or more attrs given, change tag structure.
+            for key in self.attributes: 
+                self.compose += f" {key}=\"{self.attributes[key]}\""
+            self.compose.rstrip()
+            self.compose += ">"
+            return self.compose
```

Comment:
you can condense above a bit --- it is safe to loop over empty dictionary:
```
compose = f"<{self.tag}"
for k, v in self.attributes:
    componse += f' {k}="{v}" '
compose.rstrip()
compose  += ">"
return compose
```

notice no `self` -- this is not an instance attribute, just a local variable.